From bc86c81f5ca03a27f92f20438babb8f39fe3be8c Mon Sep 17 00:00:00 2001
From: AXP-fork <noreply@localhost>
Date: Thu, 15 Oct 2026 08:37:09 +0000
Subject: [PATCH] Settings: tap to show Wi-Fi and Bluetooth MAC address

The factory Wi-Fi MAC and Bluetooth address in About phone show "Tap to show"
until tapped, so hardware identifiers don't end up in screenshots.
Uses the SettingsLib setMacSummary() hook (CalyxOS issue 3502 / AICP c8f2005).

Change-Id: I40820574f941a6283e7293c4c4da384b73d26bd1
---
 res/values/calyx_strings.xml                  |  2 ++
 res/xml/my_device_info.xml                    |  4 +--
 .../BluetoothAddressPreferenceController.java | 25 ++++++++++++++++++-
 .../WifiMacAddressPreferenceController.java   | 24 +++++++++++++++++-
 4 files changed, 51 insertions(+), 4 deletions(-)

diff --git a/res/values/calyx_strings.xml b/res/values/calyx_strings.xml
index 3676e6e..198f747 100644
--- a/res/values/calyx_strings.xml
+++ b/res/values/calyx_strings.xml
@@ -22,4 +22,6 @@
     <string name="clipboard_auto_clear_1hour">After 1 hour</string>
     <string name="clipboard_overlay_title">Show clipboard preview</string>
     <string name="clipboard_overlay_summary">Show a preview of copied content at the bottom of the screen</string>
+    <!-- AXP fork: hide hardware identifiers until tapped -->
+    <string name="device_info_tap_to_show">Tap to show</string>
 </resources>
diff --git a/res/xml/my_device_info.xml b/res/xml/my_device_info.xml
index 500cff5..7a343a1 100644
--- a/res/xml/my_device_info.xml
+++ b/res/xml/my_device_info.xml
@@ -184,7 +184,7 @@
             android:order="46"
             android:title="@string/status_device_wifi_mac_address"
             android:summary="@string/summary_placeholder"
-            android:selectable="false"
+            android:selectable="true"
             settings:enableCopying="true"/>
 
         <!-- Bluetooth address -->
@@ -193,7 +193,7 @@
             android:order="47"
             android:title="@string/status_bt_address"
             android:summary="@string/summary_placeholder"
-            android:selectable="false"
+            android:selectable="true"
             settings:enableCopying="true"/>
 
         <!-- Device up time -->
diff --git a/src/com/android/settings/deviceinfo/BluetoothAddressPreferenceController.java b/src/com/android/settings/deviceinfo/BluetoothAddressPreferenceController.java
index 431ead5..e289bd5 100644
--- a/src/com/android/settings/deviceinfo/BluetoothAddressPreferenceController.java
+++ b/src/com/android/settings/deviceinfo/BluetoothAddressPreferenceController.java
@@ -17,7 +17,11 @@
 package com.android.settings.deviceinfo;
 
 import android.content.Context;
+import android.text.TextUtils;
 
+import androidx.preference.Preference;
+
+import com.android.settings.R;
 import com.android.settings.core.PreferenceControllerMixin;
 import com.android.settingslib.core.lifecycle.Lifecycle;
 import com.android.settingslib.deviceinfo.AbstractBluetoothAddressPreferenceController;
@@ -31,5 +35,24 @@ public class BluetoothAddressPreferenceController extends
         super(context, lifecycle);
     }
 
-    // This space intentionally left blank
+    // AXP fork: don't show the hardware address until tapped, so it doesn't leak in screenshots
+    private String mAddress;
+    private boolean mRevealed;
+
+    @Override
+    protected void setMacSummary(Preference preference, String summary) {
+        mAddress = summary;
+        preference.setSummary(mRevealed ? summary
+                : mContext.getString(R.string.device_info_tap_to_show));
+    }
+
+    @Override
+    public boolean handlePreferenceTreeClick(Preference preference) {
+        if (!TextUtils.equals(preference.getKey(), getPreferenceKey()) || mAddress == null) {
+            return false;
+        }
+        mRevealed = true;
+        preference.setSummary(mAddress);
+        return true;
+    }
 }
diff --git a/src/com/android/settings/deviceinfo/WifiMacAddressPreferenceController.java b/src/com/android/settings/deviceinfo/WifiMacAddressPreferenceController.java
index c700596..a837bda 100644
--- a/src/com/android/settings/deviceinfo/WifiMacAddressPreferenceController.java
+++ b/src/com/android/settings/deviceinfo/WifiMacAddressPreferenceController.java
@@ -17,6 +17,9 @@
 package com.android.settings.deviceinfo;
 
 import android.content.Context;
+import android.text.TextUtils;
+
+import androidx.preference.Preference;
 
 import com.android.settings.R;
 import com.android.settings.core.PreferenceControllerMixin;
@@ -37,5 +40,24 @@ public class WifiMacAddressPreferenceController extends AbstractWifiMacAddressPr
         return mContext.getResources().getBoolean(R.bool.config_show_wifi_mac_address);
     }
 
-    // This space intentionally left blank
+    // AXP fork: don't show the hardware address until tapped, so it doesn't leak in screenshots
+    private String mAddress;
+    private boolean mRevealed;
+
+    @Override
+    protected void setMacSummary(Preference preference, String summary) {
+        mAddress = summary;
+        preference.setSummary(mRevealed ? summary
+                : mContext.getString(R.string.device_info_tap_to_show));
+    }
+
+    @Override
+    public boolean handlePreferenceTreeClick(Preference preference) {
+        if (!TextUtils.equals(preference.getKey(), getPreferenceKey()) || mAddress == null) {
+            return false;
+        }
+        mRevealed = true;
+        preference.setSummary(mAddress);
+        return true;
+    }
 }
-- 
2.43.0

